### PR TITLE
Fix a potentially bug.

### PR DESCRIPTION
**Pull request by monaka**
### What does this PR do?

Fix a potentially (and an obvious) bug in the deffered promise `Deffered<T>` class.

### What issues does this PR fix or reference?

N/A

#### Docs PR

N/A

### CODE REVIEW

**Comment by riuvshin** [comment]
Can one of the admins verify this patch?

**Comment by riuvshin** [comment]
Can one of the admins verify this patch?

**Comment by riuvshin** [comment]
Can one of the admins verify this patch?

**Comment by riuvshin** [comment]
Can one of the admins verify this patch?

**Comment by riuvshin** [comment]
Can one of the admins verify this patch?

**Comment by skabashnyuk** [comment]
@vitaliy-guliy @evidolob can you take a look?

File: workspace-loader/src/json-rpc/util.ts [hunk 1]
```diff
@@ -22,8 +22,8 @@ export class Deffered<T> implements IDeffered<T> {
     private rejectF;
     constructor() {
         this.promise = new Promise((resolve, reject) => {
-            this.resolve = resolve;
-            this.reject = reject;
+            this.resolveF = resolve;
```
Comment (sr229) [inline workspace-loader/src/json-rpc/util.ts:25]: I don't get why would you change this to `resolveF`, mind giving a exact reason why this definition(s) needs to be renamed as such?

**Review by sr229 — CHANGES_REQUESTED** [review]

File: workspace-loader/src/json-rpc/util.ts (hunk 1 above)
Comment (monaka) [reply workspace-loader/src/json-rpc/util.ts:25]: @sr229 Do you think this will work well,  `(new Deffered<any>(promise, reject)).resolve(undefined);` ?

**Comment by skabashnyuk** [comment]
@monaka can you tell us what is your plan regarding this pr? Do I need to ping maintainers to review this change?

File: workspace-loader/src/json-rpc/util.ts (hunk 1 above)
Comment (monaka) [reply workspace-loader/src/json-rpc/util.ts:25]: Please read the whole code again. The behavior is obvious. `(new Deffered<any>(promise, reject)).resolve(undefined);` will show `TypeError: undefined is not a function` as `resolveF` has never been initialized.

Fortunately or unfortunately, this error will not shown on the current master branch.
This is the reason why I named `Fix a *potentially* bug`

BTW, do you know `Deffered` design pattern?

**Comment by monaka** [comment]
@skabashnyuk I'm trying to fix #12063. This PR is a part of fixes.
I'm not sure my all fixes are suitable. So I separate this PR from them. The original code patched by this PR is obviously unusable.

File: workspace-loader/src/json-rpc/util.ts (hunk 1 above)
Comment (sr229) [reply workspace-loader/src/json-rpc/util.ts:25]: @monaka a bit confused but yes I got the idea. This is meant to resolve #12063 - so I believe you tested this patch?

Comment (monaka) [reply workspace-loader/src/json-rpc/util.ts:25]: The first step of resolving #12063. It needs some more patches for fixing that completely. The aim of this patch is to fix a obvious unsuitable code. An good application is from clean code, right?

I reviewed all code in Workspace-loader, run tests with Maven, traced on my Chrome. I caught no side-effect.

**Comment by sleshchenko** [comment]
ci-test

**Comment by sleshchenko** [comment]
ci-build

**Comment by vparfonov** [comment]
ci-test

**Review by sr229 — APPROVED** [review]
LGTM

**Review by vparfonov — APPROVED** [review]

**Pull request merged**